File: Programmers/CodingTestPractice/Sort/42747.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
bool comp(int a, int b){
    return a > b;
}
int solution(vector<int> citations) {
    int answer = 1;
    sort(citations.begin(), citations.end(), comp);
    if(!citations[0]) return 0;
    for(int i = 0; i < citations.size(); i++){
        if(citations[i] > i) answer++;
        else break;
    }
    return answer;
}

File: Programmers/codingTest_Kit/Sort/42747.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Integer[] arr = Arrays.stream(citations).boxed().toArray(Integer[]::new);
        Arrays.sort(arr, Collections.reverseOrder());

        if (arr[0] == 0) {
            return 0;
        }

        for (int i = 0; i < arr.length; i++) {
            if (citations[i] > i) {
                answer++;
            } else {
                break;
            }
        }

        return answer;
    }
}